### ISSUE TITLE
vagrant halt/up/destroy doesn't work due to backup issues

### ISSUE THREAD

**Issue by jonathan-dejong** [Varying-Vagrant-Vagrants/VVV#571]
Hi,

I recently experienced that vagrant just stopped working all together out of the blue. Figured I'd make sure everything was up to date so I updated vagrant, virtualbox and VVV (and vagrant plugins such as hostsupdater and trigger). 

However now on occation whenever I try to run a vagrant halt/destroy/up I end up with these errors: 
`The command "vagrant ssh -c 'vagrant_destroy'" returned a failed exit code. The
error output is shown below:
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wpsv_db.sql: No such file or directory`
(With this error for every database I have in my init-custom.sql file)
and `==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/agminvest_db.sql: No such file or directory` (also for every single file) ..

This effectively stops the commands in their tracks leaving me unable to do anything with my vagrant. All plugins are up to date etc. 

I'm just don't have the knowledge to figure out why this is happening and much less how to solve it so I'm asking you my fellow knowledgeable men and women. 

Thanks!

**Comment by jonathan-dejong** [comment]
A vagrant reload seems to fix the problems. 
Odd that it would work and not vagrant destroy?

**Comment by jeremyfelt** [comment]
Hey @jonathan-dejong, sorry for keeping you waiting!

That is strange. Do you know if the `/srv/database/backups/` directory existed before doing a `vagrant destroy`?

**Comment by jonathan-dejong** [comment]
I didn't check via SSH if that's what you're asking but it was an VVV I've setup many months ago which had run all the time before that. Like I said a vagrant reload did fix it but I'm not sure what the difference is to doing a vagrant destroy and up :)

**Comment by alex1a** [comment]
Same problem here, when i "vagrant reload" everything works fine.
Vagrant 1.7.4
Win7 x64
VVV 1.2
My Setup: https://gist.github.com/alex1a/f1ba05fd8d860e4e0bca

This error occurs after a "vagrant up" fail on SSH connection.

```bash
'Alexandre@ALEXANDRE-PC ~/vagrant-local (develop)
$ vagrant destroy
==> default: Running triggers before destroy...
==> default: Executing command "vagrant ssh -c vagrant_destroy"...
==> default: /home/vagrant/.bash_profile: line 39: /srv/www/wp-cli/utils/wp-completion.bash: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_default.sql: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_develop.sql: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_trunk.sql: No such file or directory
==> default: Command execution finished.
The command "vagrant ssh -c 'vagrant_destroy'" returned a failed exit code. The
error output is shown below:
/home/vagrant/.bash_profile: line 39: /srv/www/wp-cli/utils/wp-completion.bash: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_default.sql: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_develop.sql: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_trunk.sql: No such file or directory
```

**Comment by cjmosure** [comment]
This happened to me as well, except I would get authentication errors when trying to even `vagrant reload` (but I could ssh in with username vagrant and password vagrant). The machine would get stuck running, but I couldn't provision, halt or anything. Vagrant up and vagrant halt were giving me the database error mentioned above and vagrant reload would give me repeated authentication errors.

What ended up working for me was ssh'ing in with the username and password, not ssh key, and running the following command to add the default key (found here: https://raw.githubusercontent.com/mitchellh/vagrant/master/keys/vagrant.pub) into the authorized_keys on the machine:

```
echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > .ssh/authorized_keys
```

I also deleted the the `~/.vagrant.d/insecure_private_key` key to force regeneration of that, but I don't think that had any effect. 

I believe this may have been caused from not suspending or halting the machine before shutting down the host machine, but once I was able to cleanly halt the machine - everything was back to normal.

**Comment by defries** [comment]
Ran into the same issue as well today after a normal reboot. Reloading Vagrant was the only thing that worked.

**Comment by duikb00t** [comment]
Had the sam issue:
vagrant halt 
==> default: Running triggers before halt...
==> default: Executing command "vagrant ssh -c vagrant_halt"...
==> default: /home/vagrant/.bash_profile: line 39: /srv/www/wp-cli/utils/wp-completion.bash: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_default.sql: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_develop.sql: No such file or directory
==> default: /home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_trunk.sql: No such file or directory
==> default: Command execution finished.
The command "vagrant ssh -c 'vagrant_halt'" returned a failed exit code. The
error output is shown below:

/home/vagrant/.bash_profile: line 39: /srv/www/wp-cli/utils/wp-completion.bash: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_default.sql: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_develop.sql: No such file or directory
/home/vagrant/bin/db_backup: line 12: /srv/database/backups/wordpress_trunk.sql: No such file or directory

**Comment by khromov** [comment]
Same issue here.

**Comment by pereirinha** [comment]
I guess you guys are missing a vagrant plugin.
Try installing vagrant-triggers by running `vagrant plugin install vagrant-triggers`

**Comment by tomjn** [comment]
Is this still an issue for people using v2.1 on the `develop` branch?

**Comment by tomjn** [comment]
I'm marking this as closed, vagrant triggers is now deprecated as it was merged into VVV core in 2.1

If anybody still gets this on the very latest VVV ( 2.2.1+ ) on the latest Vagrant let me know and we'll reopen
